src/tsd/HttpQuery.java

```diff
@@ -92,24 +90,22 @@
   /** Whether or not to show stack traces in the output */
   private final boolean show_stack_trace;
 
+  /** Used to respond to png requests */
+  private final GraphHandler graphhandler;
+
   /**
    * Constructor.
    * @param request The request in this HTTP query.
    * @param chan The channel on which the request was received.
+   * @param graphhandler The graph handler to process png requests
    */
-  public HttpQuery(final TSDB tsdb, final HttpRequest request, final Channel chan) {
+  public HttpQuery(final TSDB tsdb, final HttpRequest request, final Channel chan, final LatencyStatsPlugin httplatency, GraphHandler graphhandler) {
```

Comment:
Don't change the public API here. Instead just use the ``plugin.getIntsance(name);`` method.